File: backend/feature/auth/src/main/java/com/veda/server/controller/AccountController.java
package com.veda.server.controller;

import com.veda.server.dto.request.CheckEmailRequest;
import com.veda.server.dto.request.RegisterRequest;
import com.veda.server.dto.response.AuthResponse;
import com.veda.server.dto.response.AvailabilityResponse;
import com.veda.server.service.AccountService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/accounts")
public class AccountController {

    private final AccountService accountService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public AuthResponse createAccount(@Valid @RequestBody RegisterRequest request) {
        return accountService.createAccount(request);
    }

    @PostMapping("/check-email")
    public AvailabilityResponse checkEmail(@Valid @RequestBody CheckEmailRequest request) {
        return accountService.checkEmail(request);
    }
}

File: src/main/kotlin/com/titsuko/server/controller/AccountController.kt
package com.titsuko.server.controller

import com.titsuko.server.dto.request.CheckEmailRequest
import com.titsuko.server.dto.request.RegisterRequest
import com.titsuko.server.dto.response.AccountResponse
import com.titsuko.server.dto.response.AuthResponse
import com.titsuko.server.dto.response.AvailabilityResponse
import com.titsuko.server.service.AccountService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/accounts")
class AccountController(
    private val accountService: AccountService
) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createAccount(@Valid @RequestBody request: RegisterRequest): AuthResponse {
        return accountService.register(request)
    }

    @GetMapping("/me")
    fun getProfile(): AccountResponse {
        return accountService.getProfile()
    }

    @PostMapping("/check-email")
    fun checkEmail(@Valid @RequestBody request: CheckEmailRequest): AvailabilityResponse {
        return accountService.checkEmail(request)
    }
}